exonum-java-binding/core/rust/integration_tests/tests/runtime.rs

```diff
@@ -0,0 +1,410 @@
+        runtime.get_executor(),
+        1,
+        "artifact__test_name",
+        "artifact:test-name",
+        "1.0",
+        create_service_artifact_non_instantiable_service,
+    );
+    let instances = vec![service_instance_config.0];
+
+    let test_kit_builder = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(instances);
+    assert_panics(
+        format!(
+            "{} {}",
+            "Could not find a suitable constructor in",
+            "com.exonum.binding.fakes.services.invalidservice.NonInstantiableService",
+        )
+        .as_str(),
+        || test_kit_builder.create(),
+    );
+}
+
+fn create_artifact<C>(
+    executor: &Executor,
+    artifact_name: &str,
+    artifact_version: &str,
+    build_fn: C,
+) -> (ArtifactId, DeployArguments, TempPath)
+where
+    C: FnOnce(&Executor, &str, &str) -> TempPath,
+{
+    let artifact = ArtifactId::new(
+        JavaRuntimeProxy::RUNTIME_ID,
+        format!("{}:{}", artifact_name, artifact_version),
+    )
+    .unwrap();
+    let path = build_fn(executor, &artifact.to_string(), artifact_version);
+    let deploy_args = DeployArguments {
+        artifact_filename: path.to_str().unwrap().to_string(),
+    };
+
+    (artifact, deploy_args, path)
+}
+
+fn get_service_instance_config<C>(
+    executor: &Executor,
+    instance_id: InstanceId,
+    instance_name: &str,
+    artifact_name: &str,
+    artifact_version: &str,
+    build_artifact_fn: C,
+) -> (InstanceConfig, TempPath)
+where
+    C: FnOnce(&Executor, &str, &str) -> TempPath,
+{
+    let (artifact, args, path) =
+        create_artifact(executor, artifact_name, artifact_version, build_artifact_fn);
+
+    let artifact_spec: Vec<u8> = args.to_bytes();
+    let constructor: Vec<u8> = Vec::new();
+    let instance_spec =
+        InstanceSpec::new(instance_id, instance_name, &artifact.to_string()).unwrap();
+
+    (
+        InstanceConfig::new(
+            instance_spec.clone(),
+            Some(artifact_spec.clone()),
+            constructor.clone(),
+        ),
+        path,
+    )
+}
+
+// Creates a new instance of JavaRuntimeProxy and Executor for same JVM.
+fn get_runtime(port: i32) -> JavaRuntimeProxy {
+    let runtime_config = RuntimeConfig {
+        artifacts_path: PathBuf::from("/tmp/"),
+        // Pass log4j path to avoid error messages of mis-configuration
+        log_config_path: log4j_path(),
+        port,
+        override_system_lib_path: None,
+    };
+
+    let executor = Executor::new(VM.to_owned());
+    create_service_runtime(executor, &runtime_config)
+}
+
+// Creates a new fake instance of JavaRuntimeProxy and Executor for same JVM.
+fn get_fake_runtime(facade_method: &str) -> JavaRuntimeProxy {
+    let executor = Executor::new(VM.to_owned());
+    create_fake_service_runtime_adapter(executor, facade_method)
+}
+
+fn create_blockchain(keypair: (PublicKey, SecretKey)) -> Blockchain {
```

Comment:
If I didn't miss something, you're always call this function with `gen_keypair` in place, so the argument can be removed